cvss/scorev3.go

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// scoring is implemented based on the specification at:
+// https://www.first.org/cvss/specification-document
+
+package cvss
+
+import (
+	"math"
+)
+
+// Severity represents scores severity
+type Severity int
+
+const (
+	SeverityNone Severity = iota
+	SeverityLow
+	SeverityMedium
+	SeverityHigh
+	SeverityCritical
+)
+
+func (s Severity) String() string {
+	switch s {
+	case SeverityNone:
+		return "None"
+	case SeverityLow:
+		return "Low"
+	case SeverityMedium:
+		return "Medium"
+	case SeverityHigh:
+		return "High"
+	case SeverityCritical:
+		return "Critical"
+	default:
+		panic("undefined severity")
+	}
+}
+
+// SeverityFromScore will return the severity assigned to given score
+func SeverityFromScore(score float64) Severity {
+	if score <= 0 {
+		return SeverityNone
+	}
+	if 0 < score && score < 4 {
+		return SeverityLow
+	}
+	if 4 <= score && score < 7 {
+		return SeverityMedium
+	}
+	if 7 <= score && score < 9 {
+		return SeverityHigh
+	}
+	return SeverityCritical
+}
+
+const (
+	exploitabilityCoefficient = 8.22
+	scopeCoefficient          = 1.08
+)
+
+func roundUp(x float64) float64 {
+	// round up to one decimal
+	return math.Ceil(x*10) / 10
+}
+
+// Score = combined score for the whole vector
+func (v VectorV3) Score() float64 {
+	// combines all of them
+	return v.EnvironmentalScore()
+}
+
+// BaseScore returns the base score
+func (v VectorV3) BaseScore() float64 {
+	i, e := v.ImpactScore(), v.ExploitabilityScore()
+	if i < 0 {
+		return 0
+	}
+	c := 1.0
+	if v.baseScopeChanged() {
+		c = scopeCoefficient
+	}
+
+	return roundUp(math.Min(c*(e+i), 10.0))
```

Comment:
Not a huge fan of that `math.Min()` usage, IMHO following the expression with `if x > N { x = N }` is more readable. But it's definitely a nit, up to you.

Reply:
I find math.Min more readable